src/arrayMethodSort.js

```diff
@@ -5,7 +5,36 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const length = this.length;
+
+    for (let i = 0; i < length - 1; i++) {
+      let minIndex = i;
+
+      for (let j = i + 1; j < length; j++) {
+        if (compareFunction !== undefined) {
+          if (compareFunction(this[j], this[minIndex]) < 0) {
+            minIndex = j;
+          }
+        } else {
+          if (this[j] < this[i]
+          && typeof this[0] === 'string') {
+            minIndex = j;
+          } else if (this[j].toString().localeCompare(this[i].toString()) < 0
+            && typeof this[0] === 'number') {
+            minIndex = j;
+          }
+        }
```

Comment:
here you can just call compareFunction(this[j], this[minIndex]), check value and then swap elements or not
```
[this[i], this[j]] = [this[j], this[i]];
```